csrc/type.h

```diff
@@ -488,6 +488,14 @@ inline bool hasCompatibleDataType(
     }
     auto ptr = std::get<PointerType>(dtype.type);
     return dataTypeSize(*ptr.type) == value.as<Pointer>().size();
+  } else if (std::holds_alternative<ArrayType>(dtype.type)) {
+    if (!value.is<std::vector>()) {
+      return false;
+    }
+    if (std::get<ArrayType>(dtype.type).size == 0 &&
```

Comment:
Why this is considered compatible? IIUC, we know both `dtype` and `value` are arrays, but we haven't checked the element type of each array. However, if both are empty, it seems we consider they are compatible. Why?

Reply:
For polymorphic value, an empty list has no item type. Just like for python, `[]` is just an empty list, not an "empty list of integers" or "empty list of strings"